### PR TITLE
complete kata 6 get comments by articleid

### PR DESCRIPTION
**Pull request by imaa04**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -111,3 +111,25 @@ describe("/api/articles", () => {
...
+describe("/api/articles/:article_id/comments", () => {
+  test("get all comments for a specified article", () => {
+    return request(app)
+    .get("/api/articles/1/comments")
+    .expect(200)
+    .then(({body}) => {
+      const {comments} = body;
+      expect(comments).toHaveLength(11);
+      comments.forEach((comment) => {
+        expect(comment.article_id).toBe(1);
```
Comment (johnhutcheon) [inline __tests__/app.test.js]: I would expect this test to also show that each comment has the following properties:
comment_id
votes
created_at
author
body

File: __tests__/app.test.js [hunk 2]
```diff
@@ -111,3 +111,25 @@ describe("/api/articles", () => {
...
+      });
+  });
+})
+test("GET:404 sends an appropriate status and error message when article doesn't have a comment", () => {
+        return request(app)
+          .get("/api/articles/99/comments")
+          .expect(404)
+          .then((response) => {
+            expect(response.body.msg).toBe("this article doesn't have a comment");
+          });
```
Comment (johnhutcheon) [inline __tests__/app.test.js]: Missing a few more tests here- what would happen if we sent a get request to `api/articles/banana/comments` ?
And what would we expect if its a valid ID but the article has no comments?

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Ima, just left some feedback if you could amend your testing and add the ones I've mentioned please 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -111,3 +112,49 @@ describe("/api/articles", () => {
...
+      })
+      });
+  });
+
+  test("GET:404 sends an appropriate status and error message when article id given is valid but non-existen id", () => {
+        return request(app)
+        .get("/api/articles/99/comments")
+        .expect(404)
+        .then((response) => {
+        expect(response.body.msg).toBe("this article doesn't have a comment");
```
Comment (nickkagame) [inline __tests__/app.test.js]: Dunno if this is a article that exists?  So maybe the error message should say that.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -111,3 +112,49 @@ describe("/api/articles", () => {
...
+    .get("/api/articles/banana/comments")
+    .expect(400)
+    .then((response) => {
+      expect(response.body.msg).toBe("Bad request");
+    });
+})
+  test("GET:404 sends an appropriate status and error message when given valid ID but the article has no comments", () => {
+    return request(app)
+      .get("/api/articles/7/comments")
+      .expect(404)
```
Comment (nickkagame) [inline __tests__/app.test.js]: This should be a 200

Comment (nickkagame) [reply __tests__/app.test.js]: The article exists but the article has no comments.  it is a valid response to get.

File: models/articles.models.js [hunk 5]
```diff
@@ -31,3 +31,23 @@ exports.selectArticles = () => {
     return result.rows;
   });
 };
+
+exports.selectCommentByArticleId = (article_id) => {
+  return db
+    .query(
```
Comment (nickkagame) [inline models/articles.models.js]: You need to check if the article exists in order to determine whether the empty array being returned is due to there being no comments, or due to the article not existing.  Think back to Stephens Tuesday lecture!

**Review by nickkagame — CHANGES_REQUESTED** [review]
Hi Ima,  Good testing and good work generally.  Just need to add code to check whether an article exists so that you know what error code to send back., Think back to stephen's lecture on Tuesday! He showed exactly what to do.

File: models/articles.models.js [hunk 6]
```diff
@@ -31,3 +31,34 @@ exports.selectArticles = () => {
...
+    return db
+      .query(
+        `SELECT * FROM comments WHERE article_id = $1
+        ORDER BY created_at DESC;`,
+        [article_id]
+      )
+
+      .then(( result ) => {
+        if(result.rows.length === 0) {
+          return Promise.reject({status: 200, msg: 'this article has no comments'})
```
Comment (nickkagame) [inline models/articles.models.js]: Here you can just send back the empty array for the 200 no comments. So you can remove the condition

**Review by nickkagame — APPROVED** [review]
Nice work Ima! See my comment for a small suggested edit.

**Pull request merged**